packages/notifications/push/amplify_push_notifications/lib/src/amplify_push_notifications_impl.dart

```
@@ -184,19 +207,37 @@ class AmplifyPushNotifications extends PushNotificationsPluginInterface {
     try {
       await _serviceProviderClient.registerDevice(address);
       _logger.info('Successfully registered device with the service provider');
-      // return true;
     } on Exception {
       throw const PushNotificationException(
         'Error when registering device with the service provider: ',
       );
     }
   }
 
+  Future<void> _registerCallback(
+    Function callback,
+    CallbackType callbackType,
+  ) async {
+    final callbackHandle = PluginUtilities.getCallbackHandle(callback);
+    if (callbackHandle == null) {
+      _logger.error(
+        'Callback is not a global or static function',
+      );
```

Should we thrown exception instead of/in addition to logging an error?